From patchwork Fri May  3 10:51:09 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: 'Pierre-Jean Texier' via swupdate
 <swupdate@googlegroups.com>
X-Patchwork-Id: 1094814
Return-Path: <swupdate+bncBDSMNWXTYUPRBYF2WDTAKGQEK2754HI@googlegroups.com>
X-Original-To: incoming@patchwork.ozlabs.org
Delivered-To: patchwork-incoming@bilbo.ozlabs.org
Authentication-Results: ozlabs.org;
 spf=pass (mailfrom) smtp.mailfrom=googlegroups.com
 (client-ip=2a00:1450:4864:20::53d;
 helo=mail-ed1-x53d.google.com;
 envelope-from=swupdate+bncbdsmnwxtyuprbyf2wdtakgqek2754hi@googlegroups.com;
 receiver=<UNKNOWN>)
Authentication-Results: ozlabs.org; dmarc=pass (p=none dis=none)
 header.from=googlegroups.com
Authentication-Results: ozlabs.org; dkim=pass (2048-bit key;
 unprotected) header.d=googlegroups.com header.i=@googlegroups.com
 header.b="M+mAkP1f"; dkim-atps=neutral
Received: from mail-ed1-x53d.google.com (mail-ed1-x53d.google.com
 [IPv6:2a00:1450:4864:20::53d])
 (using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
 key-exchange X25519 server-signature RSA-PSS (4096 bits)
 server-digest SHA256) (No client certificate requested)
 by ozlabs.org (Postfix) with ESMTPS id 44wTVH3sWnz9s9G
 for <incoming@patchwork.ozlabs.org>;
 Fri,  3 May 2019 20:52:21 +1000 (AEST)
Received: by mail-ed1-x53d.google.com with SMTP id 18sf3377699edw.6
 for <incoming@patchwork.ozlabs.org>;
 Fri, 03 May 2019 03:52:21 -0700 (PDT)
ARC-Seal: i=2; a=rsa-sha256; t=1556880736; cv=pass;
 d=google.com; s=arc-20160816;
 b=veUo0tY1ZgpNWRFPfXRtUhojy33zz1vwX7DKA0Kn0jmUIcaPYuM/ZX9GmL1HLB5Evd
 XKgPuywEa6K4aN8mjaWxv2zJfmGmffei12O8O9i+tcXzqus7bpYwV+yuOl84efdlN+CZ
 gqePbBzslqIdJQw3MC7gUaR/JXP35zJlyemFBb0FjG0A25fjPw45eZl+RuDkH+pJckyV
 /0a3V1W5U75aW7IYM7IAEG4NDsfqrBP10JyRz4ILIFFqH4//tDdrK9NSbyappTL9lYcX
 x6CbLtpptXjCh8TsrAw6SLzUXyd92ixwDTtPYApxoLFHTI35XpR43z6ntPB68iG31Ei5
 PvdQ==
ARC-Message-Signature: i=2; a=rsa-sha256; c=relaxed/relaxed; d=google.com;
 s=arc-20160816; 
 h=list-unsubscribe:list-subscribe:list-archive:list-help:list-post
 :list-id:mailing-list:precedence:reply-to:message-id:date:subject:cc
 :to:from:mime-version:dkim-signature;
 bh=CMSlL8nI1cOY0qnamVNiJQFLXY9BLZTdCxxBgu0CNL8=;
 b=VQ5bznDaHf1pbRud+fWBF5z+v466OB/75H3uf7eYmslaSI6XED/QCFNEMwDOm0NnsJ
 BlJEMYuv5VQhQpqf45qz2JtNTU1CeqhuXIcw1HB3amfIRwWQ+6+sJ94cSGPNbTl0N84O
 HgzyaMzO3LznXlWjIP39A0uV3QkUGC/YPxjKwOkHBxtKRqCTJUqUTPO/654m/6C6UUXr
 k0dTX6FI79RfvaP0rhymgCxOrdCH8o4pqSganROZm2tTWcXew6A93aqgtl9jdAhdpeK1
 CR4P76gqVQWrF8nJ6jOz8NkmORLr6hYcKwS3ZLRXpC8p00joT4ftSS9QaI4wppgv52q5
 /2iA==
ARC-Authentication-Results: i=2; gmr-mx.google.com;
 dkim=pass header.i=@koncepto.io header.s=default header.b=pTmHmJXo;
 spf=pass (google.com: domain of pjtexier@koncepto.io designates
 195.154.119.111 as permitted sender)
 smtp.mailfrom=pjtexier@koncepto.io; 
 dmarc=pass (p=QUARANTINE sp=QUARANTINE dis=NONE)
 header.from=koncepto.io
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=googlegroups.com; s=20161025;
 h=mime-version:from:to:cc:subject:date:message-id:x-original-sender
 :x-original-authentication-results:reply-to:precedence:mailing-list
 :list-id:list-post:list-help:list-archive:list-subscribe
 :list-unsubscribe;
 bh=CMSlL8nI1cOY0qnamVNiJQFLXY9BLZTdCxxBgu0CNL8=;
 b=M+mAkP1fflfyna/Ou8/xHZ2ERGK2ho2HbaKc4ZVpv3qvc+m6vAU2W5a3ZZhs9+jIJp
 zyuhc4Q/CBHjJONzZdGZYIivQ17hX0GLmbHudFHkzgKGjQC2i/s92zmf+qspDdck4FcX
 apiG71fUml5L7wsp7j12hYlEQFcx+TmFfBFvX9LeVEHGDTSJ+sITHeqa/3PxWKVjLc5B
 tg9aN5TQxTuCkqulD7/s0RZfrmnKdaEA33efVXkgn4YKFAN1rgmG9q+b2/vQj6i5Ur1W
 pwVVrKm/RmUY2wkdt57waGZ1RpNUror0tfSCmkCRMgS76iit3ppo77VicKA1O9pT1U1O
 huJw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:mime-version:from:to:cc:subject:date:message-id
 :x-original-sender:x-original-authentication-results:reply-to
 :precedence:mailing-list:list-id:x-spam-checked-in-group:list-post
 :list-help:list-archive:list-subscribe:list-unsubscribe;
 bh=CMSlL8nI1cOY0qnamVNiJQFLXY9BLZTdCxxBgu0CNL8=;
 b=tbdH7KPyKCjXSxmxLH5qro0y2z3873qRZb9jZJaamFS8SG765EqNcfKjZKo4imDjYg
 1/JkurkUExxRrMW4x0aSNjqzWPHAhq7FY0ZS/AzSyHjOhcC7pEDqgXRnwlYNVpbv5nz3
 UE0NMrvX77VqBu4K2LymouYkHMZthRBQeyoD9F4nZB0WjCiSL5/WmhkvU4UlCEoTKEqA
 TQb9Z7UwgWDnd4Ff1Zdf1OgHGTYq5Gatso2QBhcWyD1HO9x7PNcy2PsbYGQNLzG9h2ku
 sbORVzvzc82xhe6lrpGHXJFrOBttzKSuucgrs8I03LQukzLbPKxjdVavKd/nvVEIQ83e
 DnpQ==
X-Gm-Message-State: APjAAAW5O8EQzfJxX4eCeZ4YEfWNz75okEaCUy5uwohtafwzQMHokEs2
 efbUWalkOtZ1QQnSqqj/tDA=
X-Google-Smtp-Source: APXvYqxkUqqHvQ0FNkb0woTSubUMcr9Bia6BIF8XHP08wowbaPb2EUHA2FaAD+Gkd0NhRrCRYLWP8Q==
X-Received: by 2002:a17:906:fccb:: with SMTP id
 qx11mr5514473ejb.276.1556880736138; 
 Fri, 03 May 2019 03:52:16 -0700 (PDT)
MIME-Version: 1.0
X-BeenThere: swupdate@googlegroups.com
Received: by 2002:a17:906:3442:: with SMTP id d2ls858334ejb.1.gmail; Fri, 03
 May 2019 03:52:15 -0700 (PDT)
X-Received: by 2002:a17:906:2447:: with SMTP id
 a7mr5557892ejb.235.1556880735647; 
 Fri, 03 May 2019 03:52:15 -0700 (PDT)
ARC-Seal: i=1; a=rsa-sha256; t=1556880735; cv=none;
 d=google.com; s=arc-20160816;
 b=gQNn4O3dHDPPe3WaKPgJXFFMLGtoHTS/qscCVo6hcif2r1bcqwa6qtm0omyZaxxG9G
 elLQUdvI9ThE9w686O6NzLCfBU5mcsOTiitKPb8hl+4ZtWP5/fioS3ohUPIOreupanMH
 7cETWgJsz1PaHdprQcb+E5CiMd4iS9eHhj9cCnKZHhTDv5owwteqNCLnPeMKmfPwpwqV
 so2MPo87iWCQyTZE/H0wR3WtOWapeBOuxMWSy4VEpz750vDoADl9LGCNnWRfudVi7jK6
 +DROQqOHQpp6Nmj9i77Cd7oaJa1YIBqDukemwS/3W0NYLNVe3yvuUzr8BRDUcTcFNSQt
 gzEw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=google.com;
 s=arc-20160816; 
 h=message-id:date:subject:cc:to:from:dkim-signature;
 bh=I0xKZguFNl01eYansYeG2ChRnbek9ek0AUpf0uwU6uQ=;
 b=OHPLrlz4jcyrwMEsCWx/0gvVHKP/qw51k7Ho1ydeMftqPHug6S10m/UDWLK7cyZHhm
 nlmuMpRnlUk6dWNM1SzD1QozbVIhpJGOmshHVz0dcHbhOWOeJg9630r2ozGAOXYX9eD2
 M74x0NwsOAYciVbRl16CkmwrKXa20TzU66TD6JF+LEuSpPnvxEVx3AC9LFW8mRpb+2vm
 rTPnqz+yg2OtspODZCdxaOHKah1f8TOUTIJGyNfv9RENUY9n3SikH1sthUhYdzYcshoD
 JZ0tA52QJWL/M28DOZrDdHFgUMLOUksrFWdf3IRUCBAI+CK8Dycn+b0192MITJ4xddo/
 Y7Dw==
ARC-Authentication-Results: i=1; gmr-mx.google.com;
 dkim=pass header.i=@koncepto.io header.s=default header.b=pTmHmJXo;
 spf=pass (google.com: domain of pjtexier@koncepto.io designates
 195.154.119.111 as permitted sender)
 smtp.mailfrom=pjtexier@koncepto.io; 
 dmarc=pass (p=QUARANTINE sp=QUARANTINE dis=NONE)
 header.from=koncepto.io
Received: from koncepto.io (koncepto.io. [195.154.119.111])
 by gmr-mx.google.com with ESMTPS id
 w5si113982edw.1.2019.05.03.03.52.15 for <swupdate@googlegroups.com>
 (version=TLS1_2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256/256);
 Fri, 03 May 2019 03:52:15 -0700 (PDT)
Received-SPF: pass (google.com: domain of pjtexier@koncepto.io designates
 195.154.119.111 as permitted sender) client-ip=195.154.119.111; 
Received: from localhost.localdomain (lfbn-1-8938-37.w193-250.abo.wanadoo.fr
 [193.250.79.37]) by koncepto.io (Postfix) with ESMTPSA id 2D3DE600BF;
 Fri,  3 May 2019 12:52:15 +0200 (CEST)
From: "'Pierre-Jean Texier' via swupdate" <swupdate@googlegroups.com>
To: texier.pj2@gmail.com
Cc: Pierre-Jean Texier' via swupdate <swupdate@googlegroups.com>,
 Pierre-Jean Texier <pjtexier@koncepto.io>
Subject: [swupdate] [libubootenv][PATCH v3] fw_printenv: set
 'u-boot-initial-env' file as default
Date: Fri,  3 May 2019 12:51:09 +0200
Message-Id: <1556880669-25305-1-git-send-email-pjtexier@koncepto.io>
X-Mailer: git-send-email 2.7.4
X-Original-Sender: pjtexier@koncepto.io
X-Original-Authentication-Results: gmr-mx.google.com;       dkim=pass
 header.i=@koncepto.io header.s=default header.b=pTmHmJXo; spf=pass
 (google.com: domain of pjtexier@koncepto.io designates
 195.154.119.111 as
 permitted sender) smtp.mailfrom=pjtexier@koncepto.io; dmarc=pass
 (p=QUARANTINE sp=QUARANTINE dis=NONE) header.from=koncepto.io
X-Original-From: Pierre-Jean Texier <pjtexier@koncepto.io>
Reply-To: Pierre-Jean Texier <pjtexier@koncepto.io>
Content-Type: text/plain; charset="UTF-8"
Precedence: list
Mailing-list: list swupdate@googlegroups.com;
 contact swupdate+owners@googlegroups.com
List-ID: <swupdate.googlegroups.com>
X-Spam-Checked-In-Group: swupdate@googlegroups.com
X-Google-Group-Id: 605343134186
List-Post: <https://groups.google.com/group/swupdate/post>,
 <mailto:swupdate@googlegroups.com>
List-Help: <https://groups.google.com/support/>,
 <mailto:swupdate+help@googlegroups.com>
List-Archive: <https://groups.google.com/group/swupdate
List-Subscribe: <https://groups.google.com/group/swupdate/subscribe>,
 <mailto:swupdate+subscribe@googlegroups.com>
List-Unsubscribe: <mailto:googlegroups-manage+605343134186+unsubscribe@googlegroups.com>, 
 <https://groups.google.com/group/swupdate/subscribe>

From: Pierre-Jean Texier' via swupdate <swupdate@googlegroups.com>

Since commit [1] in U-Boot, this is the file with the initial environment
delivered with the bootloader.
So, let's set this name as default.

tested with buildroot as follows:

1) # fw_printenv
Cannot read environment, using default
baudrate=115200
...

2) # fw_setenv foo bar
Cannot read environment, using default

3) # fw_printenv foo
foo=bar

[1] https://github.com/u-boot/u-boot/commit/bdaa73a5b3923257add182b4ab8058dbfa33421b#diff-b67911656ef5d18c4ae36cb6741b7965

Signed-off-by: Pierre-Jean Texier <pjtexier@koncepto.io>
Tested-by: Joris Offouga <offougajoris@gmail.com>
---
Changes v1 -> v2:
	- fix commit log
	
Changes v2 -> v3:
	- fix commit log :(
	
 src/fw_printenv.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/fw_printenv.c b/src/fw_printenv.c
index 361d150..5f75db7 100644
--- a/src/fw_printenv.c
+++ b/src/fw_printenv.c
@@ -121,6 +121,9 @@ int main (int argc, char **argv) {
 		exit (ret);
 	}
 
+	if (!defenvfile)
+		defenvfile = "/etc/u-boot-initial-env";
+
 	if ((ret = libuboot_open(ctx)) < 0) {
 		fprintf(stderr, "Cannot read environment, using default\n");
 		if ((ret = libuboot_load_file(ctx, defenvfile)) < 0) {
